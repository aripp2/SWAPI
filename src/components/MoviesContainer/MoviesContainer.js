import React from 'react';
import './MoviesContainer.scss';
import MovieCard from '../MovieCard/MovieCard';

const MoviesContainer = ({ movies, getCharacters }) => {
  let sortedMovies = movies.sort((a, b) => a.episode_id - b.episode_id)
 console.log(movies)
  const allMovies = sortedMovies.map((movie, i) => {
    return <MovieCard
      key={i}
      id={movie.episode_id}
      title={movie.title}
      date={movie.release_date}
      characters={movie.characters}
      getCharacters={getCharacters}
    />
  })
  return (
      <section className="MovieContainer">
        { allMovies }
      </section>
    )
}

export default MoviesContainer;

