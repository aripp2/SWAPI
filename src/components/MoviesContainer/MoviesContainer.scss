.MovieContainer{
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  width: 80%;
  height: 100vh;
  margin: 15px;
}  