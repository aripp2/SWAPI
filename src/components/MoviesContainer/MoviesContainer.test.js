import React from 'react'; 
import { shallow } from 'enzyme';
import MoviesContainer from './MoviesContainer'

describe('MoviesContainer', () => {

  it('should match Snapshot', () => {
    const wrapper = shallow(
      <MoviesContainer movies={[{
        characters:["https://swapi.co/api/people/2/", "https://swapi.co/api/people/3/", "https://swapi.co/api/people/10/", "https://swapi.co/api/people/11/"],
        episode_id: 1,
        opening_crawl: "Turmoil has engulfed the Galactic Republic.The taxation of trade routes to outlying star systems is in dispute. ",
        release_date: "1999-05-19",
        title: "The Phantom Menace"
      },
      {
        characters:["https://swapi.co/api/people/2/", "https://swapi.co/api/people/3/", "https://swapi.co/api/people/6/"],
        episode_id: 2,
        opening_crawl: "There is unrest in the Galactic Senate.",
        release_date: "2002-05-16",
        title: "Attack of the Clones"
        }]} 
        getCharacters={jest.fn()} />

    )
    expect(wrapper).toMatchSnapshot();
  })


});