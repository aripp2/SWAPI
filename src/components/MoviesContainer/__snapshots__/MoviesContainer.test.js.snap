// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesContainer should match Snapshot 1`] = `
<section
  className="MovieContainer"
>
  <MovieCard
    date="1999-05-19"
    getCharacters={[MockFunction]}
    id={1}
    key="0"
    title="The Phantom Menace"
  />
  <MovieCard
    date="2002-05-16"
    getCharacters={[MockFunction]}
    id={2}
    key="1"
    title="Attack of the Clones"
  />
</section>
`;
